apps/src/studio/studio.js

```diff
@@ -521,6 +476,73 @@ function overlappingTest(x1, x2, xVariance, y1, y2, yVariance) {
   return (Math.abs(x1 - x2) < xVariance) && (Math.abs(y1 - y2) < yVariance);
 }
 
+Studio.allGoals_ = function () {
+  return Studio.spriteGoals_.concat(Studio.dynamicSpriteGoals_);
+};
+
+Studio.drawGoal = function (i, goal) {
```

Comment:
You may want to add a comment before the function to remind the caller that they will typically need to call `sortDrawOrder` afterwards. Normally, I'd suggest just making that part of the function, but it really doesn't make sense for the initialization path.